pkg/server/endpoints/authorized_entryfetcher_attested_nodes.go

```diff
@@ -24,227 +23,234 @@ type attestedNodes struct {
 	ds      datastore.DataStore
 	log     logrus.FieldLogger
 	metrics telemetry.Metrics
-	mu      sync.RWMutex
-
-	firstEventID            uint
-	firstEventTime          time.Time
-	lastEventID             uint
-	missedEvents            map[uint]time.Time
-	seenMissedStartupEvents map[uint]struct{}
-	sqlTransactionTimeout   time.Duration
+
+	eventsBeforeFirst map[uint]struct{}
+
+	firstEvent     uint
+	firstEventTime time.Time
+	lastEvent      uint
+
+	eventTracker          *eventTracker
+	sqlTransactionTimeout time.Duration
+
+	fetchNodes map[string]struct{}
+
+	// metrics change detection
+	skippedNodeEvents int
+	lastCacheStats    authorizedentries.CacheStats
 }
 
-// buildAttestedNodesCache fetches all attested nodes and adds the unexpired ones to the cache.
-// It runs once at startup.
-func buildAttestedNodesCache(ctx context.Context, log logrus.FieldLogger, metrics telemetry.Metrics, ds datastore.DataStore, clk clock.Clock, cache *authorizedentries.Cache, sqlTransactionTimeout time.Duration) (*attestedNodes, error) {
-	resp, err := ds.ListAttestedNodesEvents(ctx, &datastore.ListAttestedNodesEventsRequest{})
-	if err != nil {
-		return nil, err
+func (a *attestedNodes) captureChangedNodes(ctx context.Context) error {
+	// first, reset what we might fetch
+	a.fetchNodes = make(map[string]struct{})
```

Comment:
Does it make more sense here to zero out iff one of the below two method calls actually find anything? 

Scenario: We have fetchNodes loaded after startup. scanForNewEvents errors due to DB error. If we don't re-make the fetchNodes at the beginning, then the cache and spire-server could have continued operations.